### PR TITLE
feat(components/google-cloud): Write proper GCP_RESOURCE parameter to the custom_job output

### PR DESCRIPTION
**Pull request by SinaChavoshi**
Using the new GCP_RESOURCE proto for output of custom_job.

### CODE REVIEW

File: components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/custom_job_remote_runner.py [hunk 1]
```diff
@@ -84,9 +98,12 @@ def create_custom_job(
...
 
-    # Write the job id to output
-    with open(gcp_resources, 'w') as f:
-        f.write(custom_job_name)
+        # Write the job proto to output
+        custom_job_resource.resource_type = "CustomJob"
+        custom_job_resource.resource_uri = f"{custom_job_uri_prefix}{custom_job_name}"
+
+        with open(gcp_resources, 'w') as f:
+            f.write(json_format.MessageToJson(custom_job_resource))
```
Comment (IronPan) [inline components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/custom_job_remote_runner.py:106]: can this be unit tested somehow

Comment (SinaChavoshi) [reply components/google-cloud/google_cloud_pipeline_components/experimental/remote/gcp_launcher/custom_job_remote_runner.py:106]: Done. Good call! Thank you.

**Pull request merged**